### PR TITLE
DOCSP-30347: Convert Manage Users landing page

### PR DESCRIPTION
**Pull request by cbullinger**
## Pull Request Info

- Convert existing [Manage Users](https://www.mongodb.com/docs/realm/sdk/kotlin/users/) landing page to a container page
- Move all relevant information to any child pages 
- Update and rename existing [Authenticate Users](https://www.mongodb.com/docs/realm/sdk/kotlin/users/authenticate-users/) page to Create and Authenticate Users
- Split existing [Create and Delete Users](https://www.mongodb.com/docs/realm/sdk/kotlin/users/delete-users/) page to only Delete Users 
- Clean up any example in test suite

### Jira

- https://jira.mongodb.org/browse/DOCSP-30347

### Staged Changes

- [Create & Authenticate Users - Kotlin SDK](https://preview-mongodbcbullinger.gatsbyjs.io/realm/30347-manage-users-page/sdk/kotlin/users/authenticate-users/) - updated 
- [Delete Users - Kotlin SDK](https://preview-mongodbcbullinger.gatsbyjs.io/realm/30347-manage-users-page/sdk/kotlin/users/delete-users/) - updated and moved in TOC
- [Manage Custom User Data - Kotlin SDK](https://preview-mongodbcbullinger.gatsbyjs.io/realm/30347-manage-users-page/sdk/kotlin/users/custom-user-data/) - updated title to verb
- [Manage User Metadata - Kotlin SDK](https://preview-mongodbcbullinger.gatsbyjs.io/realm/30347-manage-users-page/sdk/kotlin/users/user-metadata/) - updated title to verb
- [Manage Users - Kotlin SDK](https://preview-mongodbcbullinger.gatsbyjs.io/realm/30347-manage-users-page/sdk/kotlin/users/) - updated; now a container

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [x] Create Jira ticket for corresponding docs-app-services update(s), if any
- [x] Checked/updated Admin API
- [x] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: examples/kotlin/shared/src/commonTest/kotlin/com/mongodb/realm/realmkmmapp/AuthenticationTest.kt [hunk 1]
```diff
@@ -28,160 +28,83 @@ import kotlin.test.*
 class AuthenticationTest: RealmTest() {
 
     private val hideNotReusingAnonymousUser = Credentials.anonymous(reuseExisting = false)
+    private val email = getRandom()
+    private val password = getRandom()
 
+
+    /*
+    ** Tests for Create and Authenticate Users page **
```
Comment (cbullinger) [inline examples/kotlin/shared/src/commonTest/kotlin/com/mongodb/realm/realmkmmapp/AuthenticationTest.kt:36]: rearranged tests in file based on page they're used on

File: source/sdk/kotlin/users/custom-user-data.txt [hunk 2]
```diff
@@ -29,22 +22,37 @@ including examples, refer to :ref:`<kotlin-ejson-encoding-for-atlas>`.
...
+
+.. versionadded:: 1.9.0
+
+You can serialize custom user data using an EJSON encoder. For more information,
+including examples, refer to :ref:`<kotlin-ejson-encoding-for-atlas>`.
+
+Prerequisites
+-------------
+
+To use custom user data, you must have an you must have an
```
Comment (lindseymoore) [inline source/sdk/kotlin/users/custom-user-data.txt:38]:
```suggestion
To use custom user data, you must have an 
```

File: source/sdk/kotlin/users/custom-user-data.txt [hunk 3]
```diff
@@ -29,22 +22,37 @@ including examples, refer to :ref:`<kotlin-ejson-encoding-for-atlas>`.
...
+App Services App with custom user data enabled. 
 
-To use custom user data, you must first enable custom user data
-in App Services:
+To set up an App Services App that uses custom user data, refer to the 
+following:
 
-#. :ref:`Create an App <create-a-realm-app>`.
-#. :ref:`Enable custom user data <enable-custom-user-data-procedure>`.
+#. :ref:`Connect to Atlas App Services <kotlin-connect-to-backend>`
```
Comment (lindseymoore) [inline source/sdk/kotlin/users/custom-user-data.txt:44]:
```suggestion
#. :ref:`Connect your app to Atlas App Services <kotlin-connect-to-backend>`
``` 
To make it a clearer procedural prerequisite step.

File: source/sdk/kotlin/users/delete-users.txt [hunk 4]
```diff
@@ -13,19 +13,8 @@ Create and Delete Users - Kotlin SDK
...
-
-For most :ref:`authentication methods <authentication-providers>`, Atlas App Services 
-automatically creates a :ref:`user object <user-objects>` the first time 
-a user authenticates. The only exception is email/password authentication. 
-When you use email/password authentication, you must :ref:`register 
-<kotlin-register-email-password-user>` and :ref:`confirm 
-<kotlin-confirm-a-new-users-email-address>` a user before the user can 
-authenticate to an App Services App.
+This page describes how to remove and delete authenticated users from 
+your Atlas App Services App. Learn how to :ref:`<kotlin-authenticate-users>`.
```
Comment (lindseymoore) [inline source/sdk/kotlin/users/delete-users.txt:17]:
```suggestion
your Atlas App Services App. For more information on how to create and authenticate users, see :ref:`<kotlin-authenticate-users>`.
```
Did you mean to do something like this to introduce further reading?

File: source/sdk/kotlin/users/delete-users.txt [hunk 5]
```diff
@@ -34,48 +23,50 @@ authenticate to an App Services App.
...
-method on a user to remove the ``User`` object from your client app. 
-This method *does not* delete the ``User`` object from the Atlas App Services App.
+To remove a ``User`` object from your client app, call the `remove 
+<{+kotlin-sync-prefix+}io.realm.kotlin.mongodb/-user/remove.html>`__
+method on a user: 
+
+.. literalinclude:: /examples/generated/kotlin/AuthenticationTest.snippet.remove-user.kt
+   :language: kotlin
 
 The ``user.remove()`` method:
```
Comment (lindseymoore) [inline source/sdk/kotlin/users/delete-users.txt:33]:
```suggestion
The ``user.remove()`` method acts as follows: 
```
Style Guide [Lists section](https://www.mongodb.com/docs/meta/style-guide/style/lists/) says lists should start with capital letter. Reformatting to help.

File: source/sdk/kotlin/users/delete-users.txt [hunk 6]
```diff
@@ -34,48 +23,50 @@ authenticate to an App Services App.
...
+
+.. literalinclude:: /examples/generated/kotlin/AuthenticationTest.snippet.remove-user.kt
+   :language: kotlin
 
 The ``user.remove()`` method: 
 
-- :ref:`Logs the user <kotlin-logout>` out if they are logged in.
-- Deletes synced realms associated with the user from the device.
-- Sets the
+- *does not* delete the ``User`` object from the Atlas App Services App
```
Comment (lindseymoore) [inline source/sdk/kotlin/users/delete-users.txt:35]:
```suggestion
- *Does not* delete the ``User`` object from the Atlas App Services App.
```

File: source/sdk/kotlin/users/delete-users.txt [hunk 7]
```diff
@@ -34,48 +23,50 @@ authenticate to an App Services App.
...
+- :ref:`logs the user <kotlin-logout>` out if they are logged in
+- deletes synced realms associated with the user from the device
```
Comment (lindseymoore) [inline source/sdk/kotlin/users/delete-users.txt:37]:
```suggestion
- Deletes synced realms associated with the user from the device.
```

File: source/sdk/kotlin/users/delete-users.txt [hunk 8]
```diff
@@ -34,48 +23,50 @@ authenticate to an App Services App.
...
+- sets the
```
Comment (lindseymoore) [inline source/sdk/kotlin/users/delete-users.txt:38]:
```suggestion
- Sets the
```

File: source/sdk/kotlin/users/delete-users.txt [hunk 9]
```diff
@@ -34,48 +23,50 @@ authenticate to an App Services App.
...
+.. literalinclude:: /examples/generated/kotlin/AuthenticationTest.snippet.remove-user.kt
+   :language: kotlin
 
 The ``user.remove()`` method: 
 
-- :ref:`Logs the user <kotlin-logout>` out if they are logged in.
-- Deletes synced realms associated with the user from the device.
-- Sets the
+- *does not* delete the ``User`` object from the Atlas App Services App
+- :ref:`logs the user <kotlin-logout>` out if they are logged in
```
Comment (lindseymoore) [inline source/sdk/kotlin/users/delete-users.txt:36]:
```suggestion
- :ref:`Logs the user <kotlin-logout>` out if they are logged in.
```

File: source/sdk/kotlin/users/delete-users.txt [hunk 10]
```diff
@@ -34,48 +23,50 @@ authenticate to an App Services App.
...
-from your client app *and* the Atlas App Services App.
+To permanently delete a ``User`` object from both your client app 
+*and* the Atlas App Services App, call the `delete 
+<{+kotlin-sync-prefix+}io.realm.kotlin.mongodb/-user/delete.html>`__ 
+method on a logged-in user:
+
+.. literalinclude:: /examples/generated/kotlin/AuthenticationTest.snippet.delete-user.kt
+   :language: kotlin
 
 The ``user.delete()`` method:
```
Comment (lindseymoore) [inline source/sdk/kotlin/users/delete-users.txt:56]:
```suggestion
The ``user.delete()`` method performs the following:
```
Same as above comment, lists should start with capital. I believe should end in period.

**Review by lindseymoore — CHANGES_REQUESTED** [review]
Some copy review comments, overall looks good so far with a few small changes needed.

File: source/sdk/kotlin/users.txt [hunk 11]
```diff
@@ -5,112 +5,19 @@ Manage Users - Kotlin SDK
...
-:ref:`user metadata <auth-provider-metadata>` , such as full name or email 
-address. When you configure these metadata fields on the App Services 
-application, you can access this metadata from you client app. 
-For more information on working with user metadata, refer to
-:ref:`Read a User's Metadata <kotlin-read-user-metadata>`. 
+   Delete Users </sdk/kotlin/users/delete-users>
+   Manage Custom User Data </sdk/kotlin/users/custom-user-data>
+   Manage User Metadata </sdk/kotlin/users/user-metadata>
+
+
```
[the remark this answers is not in the document]
Comment (cbullinger) [reply source/sdk/kotlin/users.txt:16]: Added

File: source/examples/generated/kotlin/AuthenticationTest.snippet.access-token-refresh.kt [hunk 12]
```diff
@@ -1,4 +1,3 @@
-// Gets the current refresh token for the user
```
[the remark this answers is not in the document]
Comment (cbullinger) [reply source/examples/generated/kotlin/AuthenticationTest.snippet.access-token-refresh.kt]: Re-added

File: source/sdk/kotlin/users/authenticate-users.txt [hunk 13]
```diff
@@ -187,77 +279,99 @@ code implements this flow, starting with a method call to
...
+<docs/facebook-login/android>`. 
 
-Follow the official :facebook:`Facebook Login for Android Quickstart
-<docs/facebook-login/android>` to set up the authentication flow for your
-application. In the login completion handler, get the logged in user's access
+In the login completion handler, get the logged in user's access
 token from the Facebook :facebook:`LoginResult
-<docs/reference/android/current/class/LoginResult>`. Use the access token to
-create a Facebook credential by calling `Credentials.facebook()
+<docs/reference/android/current/class/LoginResult>`.
```
[the remark this answers is not in the document]
Comment (cbullinger) [reply source/sdk/kotlin/users/authenticate-users.txt:309]: Moved to same line

File: source/sdk/kotlin/users/authenticate-users.txt [hunk 14]
```diff
@@ -187,77 +279,99 @@ code implements this flow, starting with a method call to
...
+property:
+
+.. literalinclude:: /examples/generated/kotlin/AuthenticationTest.snippet.retrieve-current-user.kt
+   :language: kotlin. 
 
-   KMM supports many environments, but this example shows sign-in on
-   the Android platform. For information about signing in with Apple
-   on Apple platforms, see the
-   :ref:`Swift SDK Example <ios-login-apple>`.
+Use this to login offline or :ref:`call an Atlas Function
```
Comment (lindseymoore) [inline source/sdk/kotlin/users/authenticate-users.txt:363]:
```suggestion
Use this to log in offline or :ref:`call an Atlas Function 
```
verb typo!

File: source/sdk/kotlin/users/delete-users.txt [hunk 15]
```diff
@@ -34,48 +23,50 @@ authenticate to an App Services App.
...
 
+The ``user.remove()`` method acts as follows: 
+
+- *Does not* delete the ``User`` object from the Atlas App Services App.
 - :ref:`Logs the user <kotlin-logout>` out if they are logged in.
 - Deletes synced realms associated with the user from the device.
 - Sets the
   `User.State <{+kotlin-sync-prefix+}io.realm.kotlin.mongodb/-user/-state/index.html>`__ 
-  to ``REMOVED``. 
+  to ``REMOVED``
```
Comment (lindseymoore) [inline source/sdk/kotlin/users/delete-users.txt:40]:
```suggestion
  to ``REMOVED``.
```

File: source/sdk/kotlin/users/delete-users.txt [hunk 16]
```diff
@@ -34,48 +23,50 @@ authenticate to an App Services App.
...
-The ``user.delete()`` method:
+.. literalinclude:: /examples/generated/kotlin/AuthenticationTest.snippet.delete-user.kt
+   :language: kotlin
 
-- Deletes synced realms associated with the user from the device.
-- Deletes the ``User`` object from the Atlas App Services server.
-- Sets the
+The ``user.delete()`` method performs the following:
+
+- deletes synced realms associated with the user from the device
```
Comment (lindseymoore) [inline source/sdk/kotlin/users/delete-users.txt:58]:
```suggestion
- Deletes synced realms associated with the user from the device.
```
Same as in above section, capitalize first word of each item in list and add period.

**Review by lindseymoore — APPROVED** [review]
Small copy things, otherwise LGTM! Thanks for work on this :)

**Pull request merged**